### ISSUE TITLE
Parser should gracefully skip rules requiring unavailable providers instead of logging errors

### ISSUE THREAD

**Issue by tsanders-rh** [konveyor/analyzer-lsp#956]
When analyzing a codebase, the rule parser logs ERROR messages for every rule that requires a provider that isn't running. This creates significant noise in the logs and confuses users.

## Current Behavior

Analyzing a TypeScript codebase (which only starts `nodejs` and `builtin` providers) produces 235+ error messages like:

```
ERRO[0007] unable to parse all the rules for ruleset error="unable to find provider for: java
unable to find provider for: java
unable to find provider for: java
...
(repeated 235+ times)
```

These errors come from default rulesets (azure, camel, eap*, etc.) that contain Java-specific rules.

## Expected Behavior

Rules requiring unavailable providers should be:
1. **Silently skipped** (or logged at DEBUG/V(5) level)
2. **Not treated as errors** - this is expected behavior when analyzing single-language codebases
3. **Allow other rules to load successfully** - partial ruleset loading should succeed

## Why This Matters

**Principle of least surprise:** When analyzing a TypeScript project, seeing hundreds of Java errors is confusing and looks like something is broken.

**Correct error classification:**
- ✅ **Skip silently:** Rule requires provider X, provider X not available
- ❌ **Error loudly:** Malformed YAML, syntax errors, invalid rule structure

**Benefits all consumers:** Any tool using analyzer-lsp (kantra, Tackle UI, custom tools) would benefit from cleaner output.

## Proposed Fix

In `parser/rule_parser.go`, when `LoadRules()` encounters a missing provider:

```go
internRuleSet, internNeedProviders, err := ruleParser.LoadRules(rulePath)
if err != nil {
    if isMissingProviderError(err) {
        // Log at debug level, skip these rules
        log.V(5).Info("skipped rules requiring unavailable provider", 
            "file", rulePath, 
            "provider", extractProviderName(err))
        // Return successfully parsed rules, ignore unparseable ones
        return partialRuleSet, availableProviders, nil
    } else {
        // Actual parsing error
        return nil, nil, err
    }
}
```

## Example Use Case

```bash
# Analyzing TypeScript codebase
kantra analyze --input /path/to/react-app

# Current: 235+ Java errors in logs
# Expected: Clean output, only relevant rules loaded
```

## Environment
- analyzer-lsp version: `v0.8.1-alpha.2` (and likely all versions)
- Issue occurs in: `parser/rule_parser.go` around line that returns `fmt.Errorf("unable to find provider for: %v", langProvider)`

## Related

This issue was discovered while testing nodejs provider integration with kantra and the new progress bar feature.
